src/main/java/com/google/sps/servlets/SearchMealServlet.java

```diff
@@ -0,0 +1,64 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.gson.Gson;
+
+@WebServlet("/search-dishes")
+public class SearchMealServlet extends HttpServlet {
+
+    private ArrayList<String> dishes;
+    private ArrayList<String> result;
+
+    @Override
+    public void init() {
+        dishes = new ArrayList<>();
+        dishes.add("fried potato with mushrooms and onion");
+        dishes.add("pizza with pineaple, sausage and tomato");
+        dishes.add("vegetable soup with onion");
+        dishes.add("chocolate cake with butter cream and strawberry");
+        dishes.add("hot chocolate with sugar and caramel");
+    }
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String gson = new Gson().toJson(result);
+
+        response.setContentType("application/json");
+        response.getWriter().println(gson);
+    }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        result = searchDishByKeyWord(getParameter(request, "text", ""));
+        
+        response.setContentType("application/json;");
+        response.getWriter().println(result);
+        response.sendRedirect("/index.html");
+    }
+    
+    private ArrayList<String> searchDishByKeyWord(String request) {
+        ArrayList<String> result = new ArrayList<String>();
```

Comment:
Nit:
I think you can use [`.filter()`](https://www.baeldung.com/java-stream-filter-lambda) method here.